webull.py

```diff
@@ -300,6 +302,83 @@ def get_quote(self, stock='') :
 
         return result
 
+    def get_analysis(self, stock=None) :
+        '''
+        get analysis info and returns a dict of analysis ratings
+        '''
+        url = f'https://securitiesapi.webullbroker.com/api/securities/ticker/v5/analysis/{self.get_ticker(stock)}'        
+        return requests.get(url).json()
+
+    def get_financials(self, stock=None) :
+        '''
+        get financials info and returns a dict of financial info
+        '''
+        url = f'https://securitiesapi.webullbroker.com/api/securities/financial/index/{self.get_ticker(stock)}'
+        return requests.get(url).json()
+
+    def get_news(self, stock=None, Id=0, items=20) :
+        '''
+        get news and returns a list of articles
+        params:
+            Id: 0 is latest news article
+            items: number of articles to return 
+        '''
+        url = f'https://securitiesapi.webullbroker.com/api/information/news/v5/tickerNews/{self.get_ticker(stock)}'
+        params = {'currentNewsId': Id, 'pageSize': items}        
+        return requests.get(url, params=params).json()
+
+    def get_options_expiration_dates(self, stock=None) :
+        '''
+        returns a list of options expiration dates
+        '''
+        url = f'https://quoteapi.webullbroker.com/api/quote/option/{self.get_ticker(stock)}/list'        
+        return requests.get(url).json()['expireDateList']
+
+    def get_options(self, stock=None, count=-1, includeWeekly=1, direction='all', expireDate=None, queryAll=0) :
+        '''
+        get options and returns a dict of options contracts
+        params:
+            stock: symbol
+            count: -1
+            includeWeekly: 0 or 1
+            direction: all, calls, puts
+            expireDate: contract expire date
+            queryAll: 0
+        '''
+        if not expireDate:
+            expireDate = self.get_options_expiration_dates(stock)[0]['date']
+        url = f'https://quoteapi.webullbroker.com/api/quote/option/{self.get_ticker(stock)}/list'
+        params = {'count': count, 'includeWeekly': includeWeekly, 'direction': direction,
+            'unSymbol': stock, 'queryAll': queryAll}
+        return requests.get(url, params=params).json()['data']
```

Comment:
I don't know if 'data' is a valid key here. You can use the following to figure out what it should be.
try:
    return requests.get(url, params=params).json()['data']
except KeyError as e:
    print(requests.get(url, params=params).json())
    print("Key error. Bad key: {}".format(e))
    return False

Reply:
Data is a valid key for the endpoint that is being called.  See the example below.
`
>>> from webull import webull
>>> webull = webull()
>>> webull.get_bars('TSLA', interval='m1', extendTrading=1) 
                       open    high     low   close  volume   vwap
timestamp                                                         
2020-01-15 08:11:00  532.55  532.55  532.28  532.28   122.0  538.0
`